src/transport/SecureSession.cpp

```diff
@@ -167,10 +171,10 @@ CHIP_ERROR SecureSession::Encrypt(const unsigned char * input, size_t input_leng
     SuccessOrExit(error);
 
     error = AES_CCM_encrypt(input, input_length, AAD, aadLen, mKey, sizeof(mKey), (const unsigned char *) IV, sizeof(IV), output,
-                            (unsigned char *) &tag, sizeof(tag));
+                            (unsigned char *) tag, taglen);
```

Comment:
Does this need a `static_assert(CHAR_BIT == 8, "Can't cast uint8_t array to char array");` or so?  Feels to me like it does.  The old code did not have this problem offhand, as long as `CHAR_BIT` was a divisor of 64, though I bet `AES_CCM_encrypt` assumes `unsigned char` is 8-bit anyway....

Reply:
could we express CHIP's crypto in unambiguous terms and handle assumptions on a per-implementation basis?

Reply:
As in, have a CHIP API that explicitly takes `uint8_t*` and then backends that assert whatever sanity they need to assert before calling into other things?  That would be nice, yes!

Reply:
yes, exactly.  put the casts down in crypto

Reply:
I agree, but it might be good to do this in a separate PR. This will make an API change that impacts tests and lot more other code, that's unrelated to this PR.

Reply:
https://github.com/project-chip/connectedhomeip/issues/2402 for tracking.

Reply:
Separate PR for API changes absolutely makes sense to me.